sapi/fpm/config.m4

```diff
@@ -489,50 +489,24 @@ if test "$PHP_FPM" != "no"; then
   if test "$PHP_FPM_ACL" != "no" ; then
     AC_CHECK_HEADERS([sys/acl.h])
 
-    AC_COMPILE_IFELSE([AC_LANG_SOURCE([[#include <sys/acl.h>
-      int main(void)
-      {
-        acl_t acl;
+    dnl *BSD has acl_* built into libc, macOS doesn't have user/group support.
+    LIBS_save="$LIBS"
+    AC_SEARCH_LIBS([acl_free], [acl],
+      [AC_MSG_CHECKING([for acl user/group permissions support])
+      AC_LINK_IFELSE([AC_LANG_PROGRAM([#include <sys/acl.h>],
+        [acl_t acl;
         acl_entry_t user, group;
         acl = acl_init(1);
         acl_create_entry(&acl, &user);
         acl_set_tag_type(user, ACL_USER);
         acl_create_entry(&acl, &group);
         acl_set_tag_type(user, ACL_GROUP);
-        acl_free(acl);
-        return 0;
-      }
-    ]])], [
-      AC_CHECK_LIB(acl, acl_free,
-        [PHP_ADD_LIBRARY(acl)
-          have_fpm_acl=yes
-        ],[
-          AC_RUN_IFELSE([AC_LANG_SOURCE([[#include <sys/acl.h>
-            int main(void)
-            {
-              acl_t acl;
-              acl_entry_t user, group;
-              acl = acl_init(1);
-              acl_create_entry(&acl, &user);
-              acl_set_tag_type(user, ACL_USER);
-              acl_create_entry(&acl, &group);
-              acl_set_tag_type(user, ACL_GROUP);
-              acl_free(acl);
-              return 0;
-            }
-          ]])],[have_fpm_acl=yes],[have_fpm_acl=no],[have_fpm_acl=no])
-        ])
-    ], [
-      have_fpm_acl=no
-    ])
-
-    AC_MSG_CHECKING([for acl user/group permissions support])
-    if test "$have_fpm_acl" = "yes"; then
-      AC_MSG_RESULT([yes])
-      AC_DEFINE([HAVE_FPM_ACL], 1, [do we have acl support?])
-    else
-      AC_MSG_RESULT([no])
-    fi
+        acl_free(acl);])],
+        [AC_MSG_RESULT([yes])
+        AC_DEFINE([HAVE_FPM_ACL], [1], [Whether FPM has acl support])],
```

Comment:
same here

Reply:
Autoconf has really strange issues with newlines and indenting these. Issue is how the expansion happens. For example, to change the above to the more logical on the first glance:

```m4
AC_SEARCH_LIBS([acl_free], [acl], [
  AC_MSG_CHECKING([for acl user/group permissions support])
  AC_LINK_IFELSE([AC_LANG_PROGRAM([#include <sys/acl.h>],
  ...
])    
```

More characters are added to the generated configure script:

```diff
diff ../configure ./configure
9326c9326,9327
<   { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for acl user/group permissions support" >&5
---
> 
>       { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for acl user/group permissions support" >&5
79532c79533
``` 

Easiest is to wrap beginning of the macro call in the argument with the `[` and put it on the new line because after that argument separator characters are trimmed.

Reply:
Why do we care about extra characters in configure? Majority of users are using distro provided packages of Docker images with pre-installed PHP. So if the configure size is a bit bigger should not really matter IMHO or do I miss anything? I think I would prefer readability in this case unless there is some really good reason not to.

Reply:
Sure, I'll update it. Otherwise, it's like a minified CSS issue in this specific case (probably). And majority of Autoconf is written in such style. With one here one there, you might get several 10 thousand lines and redundant indentation less in the configure output.

Reply:
I guess the next commit might be a bit more readable. Otherwise, what can be perhaps also done in these cases is to wrap multiple macro calls inside a single argument in `[`, `]` and make it a bit easier to handle.

```m4
AC_LINK_IFELSE([AC_LANG_PROGRAM([...], [
  ...
  ],
  [...]
  [...]
  [...],
  [...]
  [...]])
```

Writing Autoconf code for humans and to sync all these code styles in docs and php-src code is mission impossible anyway.